rust/persia-core/src/data.rs

```diff
@@ -168,40 +173,84 @@ impl PersiaBatch {
         self.inner.labels.push(tensor_impl);
     }
 
-    pub fn add_id_type_features(
+    /// Add id_type_feature which is a sparse matrix into [`PersiaBatchImpl`] with required name.
+    pub fn add_id_type_feature(
         &mut self,
-        id_type_features: Vec<(String, Vec<&PyArray1<u64>>)>,
-        requires_grad: Option<bool>,
+        id_type_feature: Vec<&PyArray1<u64>>,
+        id_type_feature_name: String,
     ) {
+        let data = vec![1, 2, 3];
+        std::mem::forget(data);
+
+        let indices = id_type_feature
+            .iter()
+            .map(|x| {
+                x.readonly()
+                    .as_slice()
+                    .expect("cannot read np array")
+                    .to_vec()
+            })
+            .collect();
+
+        self.id_type_features
+            .as_mut()
+            .unwrap()
+            .push(FeatureBatch::new(id_type_feature_name, indices));
+    }
+
+    /// Add id_type_feature which is a vector into [`PersiaBatchImpl`] with required name.
+    pub fn add_id_type_feature_with_single_id(
+        &mut self,
+        id_type_feature: &PyArray1<u64>,
+        id_type_feature_name: String,
+    ) {
+        let indices = id_type_feature
+            .readonly()
+            .as_slice()
+            .expect(format!("cannot read {}", &id_type_feature_name).as_str())
+            .iter()
+            .map(|x| vec![*x])
+            .collect();
+
+        self.id_type_features
+            .as_mut()
+            .unwrap()
+            .push(FeatureBatch::new(id_type_feature_name, indices))
+    }
+
+    /// Set requires_grad for [`PersiaBatchImpl`] and check it
+    /// valid or not. Finally convert the IdTypeFeatureBatch to EmbeddingTensor.
+    pub fn check_batch(&mut self, requires_grad: Option<bool>) -> PyResult<()> {
```

Comment:
`check_batch` sounds like a func that do not modify the struct, how about name it as `consume_batch`?

Reply:
sounds good to me

Reply:
`consume_batch` sounds like taken entire `PersiaBatchImpl`, but indeed the function is converted `id_type_feature_batches` to `EmbeddingTensor`, so I think `convert_id_type_features` is better.

Reply:
Sure, how about `convert_to_embedding_tensor`

Reply:
After discussion, it is better to modify `check_batch` to `converted_id_type_features2embedding_tensor`.